lib/rubocop/cop/rails/http_status.rb

```diff
@@ -0,0 +1,168 @@
+      #   redirect_to root_url, status: 200
+      #
+      #   # good
+      #   render :foo, status: :ok
+      #   render json: { foo: 'bar' }, status: :ok
+      #   render plain: 'foo/bar', status: :not_modified
+      #   redirect_to root_url, status: :moved_permanently
+      #
+      # @example `EnforcedStyle: numeric`
+      #   # bad
+      #   render :foo, status: :ok
+      #   render json: { foo: 'bar' }, status: :not_found
+      #   render plain: 'foo/bar', status: :not_modified
+      #   redirect_to root_url, status: :moved_permanently
+      #
+      #   # good
+      #   render :foo, status: 200
+      #   render json: { foo: 'bar' }, status: 404
+      #   render plain: 'foo/bar', status: 304
+      #   redirect_to root_url, status: 301
+      #
+      class HttpStatus < Cop
+        include ConfigurableEnforcedStyle
+
+        def_node_matcher :http_status, <<-PATTERN
+          (send nil? {:render :redirect_to}
+            _
+            (hash
+              (pair
+                (sym :status)
+                ${int sym})))
+        PATTERN
+
+        def_node_matcher :http_status_hash, <<-PATTERN
+          (send nil? {:render}
+            (hash
+              _
+              (pair
+                (sym :status)
+                ${int sym})))
+        PATTERN
+
+        def on_send(node)
+          http_status(node) do |ast_node|
+            checker = checker_class.new(ast_node)
+            return unless checker.offensive?
+            add_offense(checker.node, message: checker.message)
+          end
+
+          http_status_hash(node) do |ast_node|
+            checker = checker_class.new(ast_node)
+            return unless checker.offensive?
+            add_offense(checker.node, message: checker.message)
+          end
+        end
+
+        def autocorrect(node)
+          lambda do |corrector|
+            checker = checker_class.new(node)
+            corrector.replace(node.loc.expression, checker.preferred_style)
+          end
+        end
+
+        private
+
+        def checker_class
+          case style
+          when :symbolic
+            SymbolicStyleChecker
+          when :numeric
+            NumericStyleChecker
+          end
+        end
+
+        # :nodoc:
+        class SymbolicStyleChecker
+          MSG = 'Prefer `%<prefer>s` over `%<current>s` ' \
+                'to define HTTP status code.'.freeze
+
+          attr_reader :node
+          def initialize(node)
+            @node = node
+          end
+
+          def offensive?
+            !node.sym_type? && !custom_http_status_code?
+          end
+
+          def message
+            format(MSG, prefer: preferred_style, current: number.to_s)
+          end
+
+          def preferred_style
+            symbol.inspect
+          end
+
+          private
+
+          def symbol
+            ::Rack::Utils::SYMBOL_TO_STATUS_CODE.key(number)
+          end
+
+          def number
+            node.source.to_i
```

Comment:
You can use `node.children.first` instead. I think it is safer than `node.source.to_i`.

Reply:
BTW, probably we need `IntNode#value` for the purpose :laughing: